[Term Entry] Kotlin Strings .prependIndent() (#3834)
Codecademy/docs: content/kotlin/concepts/strings/terms/prependIndent/prependIndent.md

---
Title: '.prependIndent()'
Description: 'Adds indentation to a string.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Strings'
  - 'Methods'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> The **`.prependIndent()`** method in Kotlin is used to add indentation to a string. It is useful for formatting text, especially when we need to align or indent lines within a string.

Reviewer: ```suggestion
The **`.prependIndent()`** method in Kotlin is used to add indentation to a `string`. It is useful for formatting text, especially when we need to align or indent lines within a `string`.
```

## Syntax

```pseudo
fun String.prependIndent(indent: String = " "): String
```

- `String`: The string we are adding indentation to.
- `indent: String`: The indentation we want to add to the string.

## Example

> This example shows how to use the `.prependIndent()` method to add indents to a string:

Reviewer: ```suggestion
This example shows how to use the `.prependIndent()` method to add three spaces of indentation to the beginning of the `string`:
```

```kotlin
fun main() {
    val originalText = "'This is an indented string.'"

>     // Adds three spaces of indentation to the beginning of the string

Reviewer: ```suggestion
```
Let's remove it from here and add this in the intro of this code sample

    val indentedText = originalText.prependIndent("  ")
    println(indentedText)
}
```

> The code will return the following output:

Reviewer: ```suggestion
The above code will return the following output:
```
